src/x11/tray_manager.cpp

```diff
@@ -67,25 +67,31 @@ tray_manager::~tray_manager() {
   deactivate();
 }
 
-void tray_manager::setup() {
+void tray_manager::setup(string tray_module_name) {
   const config& conf = config::make();
   auto bs = conf.section();
   string position;
 
   try {
     position = conf.get(bs, "tray-position");
   } catch (const key_error& err) {
-    return m_log.info("Disabling tray manager (reason: missing `tray-position`)");
   }
 
-  if (position == "left") {
+  if (!position.empty() && !tray_module_name.empty()) {
+    m_log.warn("Tray position is manually defined(`tray-position`) and also set by tray module");
+  }
+  if (position.empty() && tray_module_name.empty()) {
+    return m_log.info("Disabling tray manager (reason: tray module not defined)");
```

Comment:
No need to change it, I'm just wondering. What is the reason behind calling return on the log call? Just to save one line?

Reply:
Looking at it again this doesn't really make sense. Wouldn't a error be better here since we're terminating the module(eg. via a module_error?)

Reply:
Hmm, before the tray would print an error if `tray-position` was empty or not set. I guess it would be better to just use `none` as the default value for `tray-position` and then we could just use the big if-else-if block just below to determine whether the tray should run or not. Then we wouldn't even need this check here.

But also, not having a tray is in itself not an error, so info logging is fine unless the user specified a wrong value for `tray-position`.